holmes/interactive.py

```diff
@@ -165,31 +634,116 @@ def run_interactive_loop(
+                elif command == SlashCommands.LAST_OUTPUT.value:
+                    handle_last_command(last_response, console, all_tool_calls_history)
+                    continue
+                elif command == SlashCommands.CLEAR.value:
+                    console.clear()
+                    continue
+                elif command == SlashCommands.CONTEXT.value:
+                    handle_context_command(messages, ai, console)
+                    continue
+                elif command.startswith(SlashCommands.SHOW.value):
+                    # Parse the command to extract tool index or name
+                    show_arg = original_input[len(SlashCommands.SHOW.value) :].strip()
+                    handle_show_command(show_arg, all_tool_calls_history, console)
+                    continue
+                elif command.startswith(SlashCommands.RUN.value):
+                    bash_command = original_input[
+                        len(SlashCommands.RUN.value) :
+                    ].strip()
+                    if not bash_command:
                         console.print(
-                            f"[bold {ERROR_COLOR}]No tool calls available from the last response.[/bold {ERROR_COLOR}]"
+                            f"[bold {ERROR_COLOR}]Usage: /run <bash_command>[/bold {ERROR_COLOR}]"
                         )
                         continue
 
-                    display_tool_calls(last_response.tool_calls, console)
+                    result = None
+                    output = ""
+                    error_message = ""
+
+                    try:
+                        console.print(
+                            f"[bold {STATUS_COLOR}]Running: {bash_command}[/bold {STATUS_COLOR}]"
+                        )
+                        result = subprocess.run(
+                            bash_command, shell=True, capture_output=True, text=True
+                        )
+
+                        output = result.stdout + result.stderr
+                        if result.returncode == 0:
+                            console.print(
+                                f"[bold green]✓ Command succeeded (exit code: {result.returncode})[/bold green]"
+                            )
+                        else:
+                            console.print(
+                                f"[bold {ERROR_COLOR}]✗ Command failed (exit code: {result.returncode})[/bold {ERROR_COLOR}]"
+                            )
+
+                        if output.strip():
+                            console.print(
+                                Panel(
+                                    output,
+                                    padding=(1, 2),
+                                    border_style="white",
+                                    title="Command Output",
+                                    title_align="left",
+                                )
+                            )
+
+                    except KeyboardInterrupt:
```

Comment:
This is great. I tried to run some command that hangs. Catching KeyboardInterrupt helps end the command without breaking the interactive mode.

Reply:
Thanks :)